### PR TITLE
Upgrade to raw-window-handle version 0.5

### PR DESCRIPTION
**Pull request by greatest-ape**
TODO:

- [x] Test on X11
- [x] Test on Windows
- [x] Test on macOS

https://github.com/greatest-ape/iced_baseview/tree/iced-0.5 might be useful for testing

### CODE REVIEW

File: src/x11/window.rs [hunk 1]
```diff
@@ -668,14 +666,25 @@ impl Window {
 
 unsafe impl HasRawWindowHandle for Window {
     fn raw_window_handle(&self) -> RawWindowHandle {
-        let mut handle = XlibHandle::empty();
+        let mut handle = XlibWindowHandle::empty();
         handle.window = self.window_id as c_ulong;
```
Comment (robbert-vdh) [inline src/x11/window.rs:670]: The window's visual is missing here: https://github.com/rust-windowing/raw-window-handle/pull/83

**Review by robbert-vdh — CHANGES_REQUESTED** [review]

File: src/gl/x11.rs [hunk 2]
```diff
@@ -80,19 +78,13 @@ impl GlContext {
...
-            return Err(GlError::InvalidWindowHandle);
-        };
-
-        if handle.display.is_null() {
+        if display.is_null() {
             return Err(GlError::InvalidWindowHandle);
         }
 
-        let display = handle.display as *mut xlib::_XDisplay;
+        let display = display as *mut xlib::_XDisplay;
```
[the remark this answers is not in the document]
Comment (greatest-ape) [reply src/gl/x11.rs]: Fixed

File: src/x11/window.rs (hunk 1 above)
Comment (greatest-ape) [reply src/x11/window.rs]: Fixed, provided that the https://docs.rs/xcb/0.9.0/xcb/xproto/type.Visualid.html used to create the window is indeed what is expected

**Comment by robbert-vdh** [comment]
Rebased against the current master branch: https://github.com/robbert-vdh/baseview/tree/raw-window-handle-0.5 (in case anyone else needs it, or someone wants to merge this PR)

**Comment by robbert-vdh** [comment]
Thanks for the PR! I rebased this on top of the latest master branch. ([diff](https://github.com/RustAudio/baseview/compare/cfb2b091120c2a7d31628802c9e76b39b5c103a1..0df43486ba43c53bb4e69775f35ef3f1569640b0)) There are still a bunch of old unmerged PRs, so I'm going through those now so we can reduce the backlog somewhat.

**Pull request merged**